client/login/controller.js

```
@@ -558,6 +558,119 @@ export async function jetpackAppleAuthCallback( context, next ) {
+	} catch {
+		context.store.dispatch( {
+			type: 'NOTICE_CREATE',
+			notice: {
+				status: 'is-error',
+				text: 'Error during GitHub authentication. Please try again.',
+			},
+		} );
+	}
+
+	// Fall back to regular login form
+	context.primary = (
+		<WPLogin isJetpack path={ context.path } query={ query } locale={ context.params.lang } />
+	);
+	next();
+}
+
+export async function jetpackGitHubAuthCallback( context, next ) {
+	const { query } = context;
+
+	const code = query.code;
+	const service = query.service;
+
+	// Not a redirect from GitHub if no code or error present
+	if ( ! code || service !== 'github' ) {
+		return next();
+	}
+
+	const redirect_to = window.sessionStorage.getItem( 'github_redirect_to' ) ?? '/';
+	window.sessionStorage.removeItem( 'github_redirect_to' );
+
+	try {
+		// GitHub supports localhost auth; and we allowlist the jetpack callback path
+		const redirectUri = `${ window.location.origin }/log-in/jetpack/github/callback`;
+
+		// Exchange auth code for tokens
+		const response = await postLoginRequest( 'exchange-social-auth-code', {
+			service: 'github',
+			auth_code: code,
+			redirect_uri: redirectUri,
+			client_id: config( 'wpcom_signup_id' ),
+			client_secret: config( 'wpcom_signup_key' ),
+		} );
+
+		const { access_token } = response.body.data;
+
+		// Try to create a new WordPress.com account (if it doesn't exist) - then, log in the user
+		try {
+			try {
+				await wpcom.req.post( '/users/social/new', {
+					service: 'github',
+					access_token,
+					signup_flow_name: 'github-auth-signup',
+					locale: getLocaleSlug(),
+					client_id: config( 'wpcom_signup_id' ),
+					client_secret: config( 'wpcom_signup_key' ),
+					tos: JSON.stringify( getToSAcceptancePayload() ),
+				} );
+			} catch {
+				// Silently fail; when id_token is not present, the endpoint fails when the user already exists
```

Is this something we'd like to log to logstash if it fails? Or perhaps check the error code if there's any, so we know we are failing silently only when user already exists